PHP 8.4 support

phpfui opened the pull request:
* Updated composer.json to support PHP 8.0 and higher only
* Moved to testing under GitHub Actions
* Ported to newer PHPUnit
* Added and corrected types for PHP 8.4 support

ToDo:
-----

Unit test need some work, but not sure what to do here as I am not familiar with the code.

phpfui committed:
Update composer.json to latest and require PHP 8.0 or higher

phpfui committed:
Convert to GitHub Actions for testing

phpfui committed:
Modernize PHPUnit

phpfui committed:
Convert to PHP 8.0

AidasK commented on src/Flow/Basic.php:
```diff
@@ -14,15 +14,13 @@ class Basic
-     * @return bool
      */
-    public static function save($destination, $config, RequestInterface $request = null)
+    public static function save(string $destination, $config, ?RequestInterface $request = null) : bool
```
maybe type config as `string|ConfigInterface`

AidasK commented on src/Flow/Config.php:
```diff
@@ -4,111 +4,86 @@
     {
         $this->config['hashNameCallback'] = $callback;
+
+		return $this;
```
not aligned?

AidasK commented on src/Flow/Config.php:
```diff
@@ -4,111 +4,86 @@
     {
         $this->config['preprocessCallback'] = $callback;
+
+		return $this;
```
align?

AidasK commented on src/Flow/Config.php:
```diff
@@ -4,111 +4,86 @@
     {
         $this->config['deleteChunksOnSave'] = $delete;
+
+		return $this;
```
code style

AidasK commented on src/Flow/Mongo/MongoConfig.php:
```diff
@@ -10,22 +10,16 @@
+    function __construct(private Bucket $gridFS)
     {
         parent::__construct();
         $this->gridFs = $gridFS;
```
remove this?

AidasK commented:
Are these tests passing locally?

phpfui commented:
Will fix these issues. Maybe I should set up phpcsfixer and phpstan too.

Tests fail for me locally, you will have to fix, as you know the code better. Not sure how the mocking works.

phpfui committed:
Run PHP-CS-Fixer with PSR2 rules

phpfui committed:
Install PHPStan and run

phpfui commented:
I fixed the save() parameter type.  Also fixed the formatting with PHP-CS-Fixer.  Corrected a bunch of type errors with PHPStan (removed the MongoConfigInterface, as it is not really needed). I also fixed one of the PHPUnit errors.  The other two are caused by the mocking, but not exactly sure how to fix those, so you will need to work on them.

I am going to test this in my project.  Will push again if I find and fix any issues.  I can't test the MongoDB stuff, as I use SQL only.

phpfui commented:
My local testing seems to work with no mods needed.  This should probably be released as v1.3 or v2.0

AidasK commented on src/Flow/Mongo/MongoFile.php:
```diff
@@ -25,69 +24,53 @@ class MongoFile extends File
             'files_id' => $this->getGridFsFile()['_id'],
-            'n' => (intval($index) - 1)
-        ]) !== null;
+            'n' => ((int) $index - 1)
```
n was probably bool before?

AidasK commented:
Released as 2.0, thank you for your contribution <3

phpfui commented:
Thanks for the prompt attention. Seems PHP 8.4 compatibility is not a high priority for many maintainers.
